### PR TITLE
BB-16: update lifecycle populator filter for metadata entries

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

**Review by rachedbenmustapha — APPROVED** [review]

File: extensions/lifecycle/LifecycleQueuePopulator.js [hunk 1]
```diff
@@ -96,28 +96,70 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
...
+        // raft log entries
+        // {
+        //   bucket: bucket name
+        //   key: no key value
+        //   value: bucket value
+        // }
+        return entry.key === undefined && entry.bucket !== METASTORE;
+    }
+
+    _isMongoEntry(entry) {
```
Comment (naren-rajendran) [inline extensions/lifecycle/LifecycleQueuePopulator.js]: Mongo entry? Is this for 7.x branches?

Comment (jonathan-gramain) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: Yes, should not be needed normally as it's 8.x specific (and should already be compatible)

File: extensions/lifecycle/LifecycleQueuePopulator.js [hunk 2]
```diff
@@ -96,28 +96,70 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
         });
     }
 
+    _isBucketdEntry(entry) {
```
Comment (jonathan-gramain) [inline extensions/lifecycle/LifecycleQueuePopulator.js]: It seems you are specifically looking for bucket change entries, so maybe rename to `isBucketEntry` (assuming you also remove the `isMongoEntry` helper)

File: extensions/lifecycle/LifecycleQueuePopulator.js (hunk 1 above)
Comment (alexanderchan-scality) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: `MongoEntry` may be a confusing name as `7.x` won't be supporting mongo. I can rename to `FileMDEntry`.

File: extensions/lifecycle/LifecycleQueuePopulator.js (hunk 2 above)
Comment (alexanderchan-scality) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: I believe the `isMongoEntry` (will rename to `isFileMDEntry`) is still needed to handle the `file` MD backend

**Review by nicolas2bert — APPROVED** [review]

File: extensions/lifecycle/LifecycleQueuePopulator.js (hunk 2 above)
Comment (jonathan-gramain) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: OK that works. But I believe the name should reflect that it's selecting bucket-update entries only, where the current name suggests it's filtering all entries coming from bucketd.

Comment (jonathan-gramain) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: What about: `isBucketEntryFromBucketd()` and `isBucketEntryFromFileMD`

**Pull request merged**